pkg/image/importer/importer.go

```diff
@@ -282,6 +282,27 @@ func applyErrorToRepository(repository *importRepository, err error) {
 	}
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image) error {
+	bs := repo.Blobs(ctx)
+
+	layerSet := sets.NewString()
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
If Docker 1.10 or newer ask for old (schema1) image manifest then it will be returned without conversion to schema2. Take a look: https://github.com/docker/distribution/blob/master/registry/handlers/images.go#L65 (`isSchema2` is false).

docker-registry-2.4 does not help if we will ask for old image. We can't force docker/distribution to return only schema2.

>  How close are we to schema2 in master?

I'm fixing tests in https://github.com/openshift/origin/pull/8938